data-access-layer/eventmesh/src/ApiServerClient.js

```diff
@@ -630,7 +631,7 @@ class ApiServerClient {
     if (opts.query) {
       query.qs = opts.query;
     }
-    const namespaceId = opts.namespaceId ? opts.namespaceId : CONST.APISERVER.DEFAULT_NAMESPACE;
+    const namespaceId = opts.namespaceId ? opts.namespaceId : CONST.APISERVER.DEFAULT_NAMESPACE; // Currently most callers are calling this function with allNamespaces: true, only metering jobs are calling without NS and defaults to. Should not be used in any other context.
```

Comment:
Should we check if `apiserver.services_namespace` is set in config?
Like

 `const namespaceId = opts.namespaceId ? opts.namespaceId : (_.get(config, 'apiserver.services_namespace') ? _.get(config, 'apiserver.services_namespace') : CONST.APISERVER.DEFAULT_NAMESPACE);`

Reply:
As I have commented in the code, currently only metering jobs are calling this to fetch the metering CRs, and they are created in default namespaces. Since this is not utilized in the interoperator, I thought of not disturbing it at this moment, I have just added the comment as warning sign. ;)